File: core/src/com/broll/gainea/server/core/goals/impl/e3/G_AllSwamps.java
package com.broll.gainea.server.core.goals.impl.e3;

import com.broll.gainea.server.core.goals.GoalDifficulty;
import com.broll.gainea.server.core.goals.OccupyGoal;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.ExpansionType;

public class G_AllSwamps extends OccupyGoal {
    public G_AllSwamps() {
        super(GoalDifficulty.MEDIUM, "Erobere alle Sümpfe");
    }

    @Override
    protected void initOccupations() {
        occupy(it -> it.getType() == AreaType.BOG, ExpansionType.BOGLANDS);
    }
}

File: server/src/main/com/broll/gainea/server/core/goals/impl/e3/G_AllSwamps.kt
package com.broll.gainea.server.core.goals.impl.e3

import com.broll.gainea.server.core.goals.GoalDifficulty
import com.broll.gainea.server.core.goals.OccupyGoal
import com.broll.gainea.server.core.map.AreaType
import com.broll.gainea.server.core.map.ExpansionType

class G_AllSwamps : OccupyGoal(GoalDifficulty.MEDIUM, "Erobere alle Sümpfe") {
    override fun initOccupations() {
        occupy({ it.type == AreaType.BOG }, ExpansionType.BOGLANDS)
    }
}
